AlgebraInLean/Basic.lean

```diff
@@ -1,40 +1,42 @@
 import Mathlib.Tactic
 
 namespace Defs
+  variable {G : Type*}
+
   class Magma (G : Type*) where
     op : G → G → G
 
   def μ [Magma G] : G → G → G := Magma.op
 
   class Semigroup (G : Type*) extends Magma G where
-    protected op_assoc : ∀ a b c : G, μ (μ a b) c = μ a (μ b c)
+    protected op_assoc : ∀ (a b c : G), μ (μ a b) c = μ a (μ b c)
 
   @[simp]
   theorem op_assoc [Semigroup G] (a b c : G) : μ (μ a b) c = μ a (μ b c) := Semigroup.op_assoc a b c
 
   class Monoid (G : Type*) extends Semigroup G where
     protected id : G
-    protected op_id : ∀ a : G, μ a id = a
-    protected id_op : ∀ a : G, μ id a = a
+    protected op_id : ∀ (a : G), μ a id = a
+    protected id_op : ∀ (a : G), μ id a = a
 
   def 𝕖 [Monoid G] : G := Monoid.id
 
   @[simp]
-  theorem op_id [Monoid M] (a : M) : μ a 𝕖 = a := Monoid.op_id a
+  theorem op_id [Monoid G] (a : G) : μ a 𝕖 = a := Monoid.op_id a
```

Comment:
Curious about the rationale behind this change. Is `G` usually used in reference to monoids?

Reply:
I just had a `variable {G : Type*}` up above and didn't feel like using different names for things. Maybe I should have used a generic name like `α` for all of them.

Reply:
Valid; the only reason I asked was for if monoids appear in the chapters. This doesn’t really matter for our definitions right now since they’ll get replaced soon.